components/src/directives/dropdown-direction/index.ts

```diff
@@ -0,0 +1,34 @@
+import {Directive, DirectiveBinding} from 'vue';
+
+function fixPosition(el: HTMLElement) {
+  el.classList.remove('--positon-top');
+  el.classList.remove('--positon-bottom');
+  if (window.innerHeight - el.getBoundingClientRect().bottom < 0) {
+    el.classList.add('--positon-top');
+    if (el.getBoundingClientRect().top < 0) {
+      el.classList.remove('--positon-top');
+      el.classList.add('--positon-bottom');
+    }
+  } else {
+    el.classList.add('--positon-bottom');
+  }
+}
+
+function handleScroll(this: any) {
+  fixPosition(this);
+}
+
+const dropdownDirectionDirective: Directive = {
+  mounted(el: HTMLElement) {
+    window.addEventListener('scroll', handleScroll.bind(el));
```

Comment:
can be refactored to
`window.addEventListener('scroll', fixPosition(el));`

and can remove `handleScroll`

Reply:
type of the arguments which are expected in 'handleScroll' listener handler function and 'fixPosition' are different.
When registering an event, we can't directly pass values to the listener handler function.